code/modules/recycling/conveyor2.dm

```diff
@@ -184,13 +211,15 @@
 	speed_process = 1
 
 
-/obj/machinery/conveyor_switch/New()
-	..()
+obj/machinery/conveyor_switch/New(newloc, newid)
+	..(newloc)
+	if(!id)
+		id = newid
 	update()
 
 	spawn(5)		// allow map load
 		conveyors = list()
-		for(var/obj/machinery/conveyor/C in world)
+		for(var/obj/machinery/conveyor/C in machines)

```

Comment:
Tried it...
 There doesn't seem to be /obj/machinery/New() in machinery.dm - so I created one, and it doesn't compile...
Actually I'll just commit what I tried. It's probably super wrong, but you'll be able to see my dumb, and point me in a better direction.